mantidproject/mantid
# Crate SQE fit functions and add to ConvFit

This PR adds three new fit functions: FickDiffusionSQE, ChudleyElliotSQE, and HallRossSQE. and adds them to ConvFit. These functions us the non SQE function to get the width to calculate the S(Q,E) fit.

**Report to:** @SpencerHowells

**To test:**

1. Open Indirect Data Analysis
2. go to the conv fit tab
3. click add workspace
4. for workspace add iris26176_graphite002_red
5. for resolution add iris26173_graphite002_res
6. check `All Workspaces`
7. click add
8. close the add worksapce windows
9. in fit type select `Fick Diffusion` and click run, check that it runs correctly.
9. in fit type select `Chudley Elliot` and click run, check that it runs correctly.
9. in fit type select `Hall Ross` and click run, check that it runs correctly.

Fixes #34190

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
